pkg/client/client.go

```diff
@@ -155,31 +163,30 @@ func NewClient(ctrlRuntimeClient client.Client, secretName, namespace, region st
 			}
 			return nil, err
 		}
-		accessKeyID, ok := secret.Data[AwsCredsSecretIDKey]
-		if !ok {
-			return nil, machineapiapierrors.InvalidMachineConfiguration("AWS credentials secret %v did not contain key %v",
-				secretName, AwsCredsSecretIDKey)
-		}
-		secretAccessKey, ok := secret.Data[AwsCredsSecretAccessKey]
-		if !ok {
-			return nil, machineapiapierrors.InvalidMachineConfiguration("AWS credentials secret %v did not contain key %v",
-				secretName, AwsCredsSecretAccessKey)
+		sharedCredsFile, err := sharedCredentialsFileFromSecret(&secret)
+		if err != nil {
```

Comment:
Looks to me like if the temp file can be created but not _written to_ this function returns early and the temp file is never deleted.

Reply:
I'm not sure of a real world situation in which this could occur but I can try to clean it up